reference/common_errors.rst

```diff
@@ -0,0 +1,143 @@
+
+We can see this in action by distributing rows across worker nodes, and then running two concurrent transactions with conflicting updates:
+
+.. code-block:: sql
+
+  CREATE TABLE lockme (id int, x int);
+  SELECT create_distributed_table('lockme', 'id');
+
+  -- id=1 goes to one worker, and id=2 another
+  INSERT INTO lockme VALUES (1,1), (2,2);
+
+  --------------- TX 1 ----------------  --------------- TX 2 ----------------
+  BEGIN;
+                                         BEGIN;
+  UPDATE lockme SET x = 3 WHERE id = 1;
+                                         UPDATE lockme SET x = 4 WHERE id = 2;
+  UPDATE lockme SET x = 3 WHERE id = 2;
+                                         UPDATE lockme SET x = 4 WHERE id = 1;
+
+::
+
+  ERROR:  40P01: canceling the transaction since it was involved in a distributed deadlock
+  LOCATION:  ProcessInterrupts, postgres.c:2988
+
+Cannot establish a new connection for placement *n*, since DML has been executed on a connection that is in use
+---------------------------------------------------------------------------------------------------------------
+
+.. code-block:: sql
+
+  BEGIN;
+  INSERT INTO http_request (site_id) VALUES (1337);
+  INSERT INTO http_request (site_id) VALUES (1338);
+  SELECT count(*) FROM http_request;
+
+::
+
+  ERROR:  25001: cannot establish a new connection for placement 314, since DML has been executed on a connection that is in use
+  LOCATION:  FindPlacementListConnection, placement_connection.c:612
+
+This is a current limitation. In a single transaction Citus does not support running insert/update statements with the :ref:`router_executor` that reference multiple shards, followed by a read query that consults both of the shards.
+
+Could not connect to server: Cannot assign requested address
+------------------------------------------------------------
+
+::
+
+  WARNING:  connection error: localhost:9703
+  DETAIL:  could not connect to server: Cannot assign requested address
+
+This occurs when there are no more sockets available by which the coordinator can respond to worker requests. It can be mitigated by re-using TCP sockets at the OS level. Execute this on the shell in the coordinator node:
+
+.. code-block:: bash
+
+  sysctl -w net.ipv4.tcp_tw_reuse=1
+
+Remaining connection slots are reserved for non-replication superuser connections
+---------------------------------------------------------------------------------
+
+(Citus Cloud error.)
+
+This occurs when Citus Cloud runs out of available connections. See :ref:`cloud_pgbouncer` to learn how to connect through Cloud's built-in PgBouncer instance to queue incoming connections.
+
+PgBouncer cannot connect to server
+----------------------------------
+
+In a self-hosted Citus cluster, this error indicates that the coordinator node is not responding to PgBouncer. Check the health of the server.
```

Comment:
Why is this for self-hosted setups? @samay-sharma Have we a lot of people using self-hosted pgBouncer? I am surprised this is a common error message. 

Also, could we elaborate a bit on the "check the health of the server"?